themes/website/static/js/osdetect.js

```diff
@@ -183,28 +197,60 @@ if (window.navigator.userAgent.indexOf("Windows Phone") != -1) {
 	All the magic stuff the makes the buttons change and stuff
 */
 
+// Set platform header
+document.getElementById("downloadHeader").innerHTML = "Download Tox for " + OSName;
+
 // Set platform toxlock
 document.getElementById("platImg").src = "theme/img/plat/" + OSName.toLowerCase() + ".svg";
 
 // Remove normal button
 document.getElementById("defaultButton").innerHTML = "";
 
+var buttonArea = document.getElementById("buttonArea");
+
+
 // Loop through all links and make buttons and info modals
-for (var i = 0; i < clients.length; i++) {
+for (var i = 0; i < clients.length; i++) 
+{
 	var client = clients[i];
 
-	var button = "\
-	<a id='link" + i + "' href='" + client.dlLink + "' class='button large-button download'>\
-		<span class='fa fa-" + client.icon + "'>&nbsp;</span>" + client.title + "\
-	</a>";
+	if (!client.icon)
+		client.icon = "";
+
+	if (client.faicon)
+		client.faicon = "fa fa-" + client.faicon;
+	else
+		client.faicon = "";
+
+	// create button
+	var button = document.createElement("A");
+	button.id = "link"+i;
+	button.href = client.dlLink;
+	button.className = "button large-button download";
+
+	// create icon
+	var buttonIcon = document.createElement("SPAN");
+	buttonIcon.className = client.faicon + " " + client.icon + " icon-distro-small";
 
-	var infoButton = "";
+	// append icon and client name to the button
+	button.appendChild(buttonIcon);
+	button.innerHTML += " " + client.title;
+
+
+	// create info button if needed
+	var infoButton;
 	if (client.desc) {
-		infoButton = "\
-		<a class='button large-button' style='box-shadow:none;color:#368CCA;' href='clients.html#" + client.name + "' title='More info'>\
-			<span class='fa fa-info-circle'></span>\
-		</a>";
+		infoButton = document.createElement("A");
+		infoButton.className = "button large-button";
+		infoButton.style = "box-shadow:none;color:#368CCA";
```

Comment:
Applying style from JS makes it difficult to troubleshot issues with CSS later, make a class for that?

Reply:
Done.